omod/src/main/java/org/bahmni/module/hwcinventory/web/controller/EsanjeevaniController.java

```diff
@@ -0,0 +1,66 @@
+package org.bahmni.module.hwcinventory.web.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.bahmni.module.hwcinventory.contract.LaunchRequest;
+import org.bahmni.module.hwcinventory.exception.LGDCodeNotFoundException;
+import org.bahmni.module.hwcinventory.service.EsanjeevaniService;
+import org.openmrs.Privilege;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Map;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/esanjeevani")
+public class EsanjeevaniController extends BaseRestController {
+
+    @Autowired
+    private EsanjeevaniService esanjeevaniService;
+
+    @RequestMapping(value = "/launch", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity<String> launchEsanjeevani(@RequestBody LaunchRequest launchRequest) {
+        Privilege privilege = Context.getUserService().getPrivilege("app:eSanjeevani");
+        if (Context.getAuthenticatedUser().getPrivileges().contains(privilege)) {
+            try {
+                String loginResponse = esanjeevaniService.getLoginResponse(launchRequest.getUsername(), launchRequest.getPassword());
+                if (esanjeevaniService.isSuccessResponse(loginResponse)) {
```

Comment:
Switch this to guard clause so we don't need else statement.

Reply:
Done